packages/babel-parser/src/plugins/typescript.js

```diff
@@ -319,6 +319,44 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       return this.finishNode(node, "TSTypeParameterDeclaration");
     }
 
+    tsTryNextParseConstantContext(): ?N.TsTypeReference {
+      if (this.lookahead().type === tt._const) {
+        this.next();
+        return this.tsParseTypeReference();
+      }
+      return null;
+    }
+
+    tsCheckLiteralForConstantContext(node: N.Node) {
+      switch (node.type) {
+        case "StringLiteral":
+        case "TemplateLiteral":
+        case "NumericLiteral":
+        case "BooleanLiteral":
+        case "SpreadElement":
+        case "ObjectMethod":
+        case "ObjectExpression":
+          return;
+        case "ArrayExpression":
+          return (node: N.ArrayExpression).elements.forEach(element => {
+            if (element) {
```

Comment:
when element might be missing? 🤔

Reply:
Maybe the second element of `[foo, ,bar]`?

Reply:
Seems like the reason 👌